stdlib/public/core/TypeDiscovery.swift

```diff
@@ -0,0 +1,105 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// A function that instantiates the type described by a type descriptor.
+///
+/// - Parameters:
+///   - descriptor: A type descriptor as passed to `_swift_enumerateTypes()`'s
+///     callback.
+///
+/// - Returns: The initialized type corresponding to `descriptor`, or `nil` if
+///   the type could not be initialized. This value can subsequently be cast to
+///   `Any.Type` using `unsafeBitCast(_:to:)`.
+private typealias _TypeGetter = @convention(c) (
+  _ descriptor: UnsafeRawPointer
+) -> UnsafeRawPointer?
+
+/// A type describing another type available in the current process.
+@available(SwiftStdlib 9999, *)
+public struct _TypeRecord {
```

Comment:
Following the "leading underscore" convention documented in [TSPL](https://docs.swift.org/swift-book/ReferenceManual/LexicalStructure.html#ID412), even though `_TypeRecord` is public it will be excluded from the standard library's documentation.  Pointing it here out to confirm that matches what you expect.

Reply:
Yep, as of right now this is not an approved addition to the Swift Standard Library. We wouldn't want to document it until/unless it went through the evolution process.